main/network.c

```diff
@@ -269,6 +270,42 @@ PHPAPI int php_network_getaddresses(const char *host, int socktype, struct socka
 }
 /* }}} */
 
+/* {{{ php_network_getaddresses
+ * Returns number of addresses, 0 for none/error
+ */
+PHPAPI int php_network_getaddresses(const char *host, int socktype, struct sockaddr ***sal, zend_string **error_string)
```

Comment:
Does it make sense to make the return value unsigend then?

Reply:
It's a little ugly since it reuses an internal variable for count and getaddrinfo return values (which is 0 or error) and would affect all the callsites, but it's possible

Reply:
Well let's leave this for a follow-up then